Source/Dafny/Translator.cs

```diff
@@ -18735,9 +18735,19 @@ public class Substituter
           foreach (var bv in s.BoundVars) {
             substMap.Remove(bv);
           }
+
           // Put things together
           var rr = new ForallStmt(s.Tok, s.EndTok, newBoundVars, SubstAttributes(s.Attributes), Substitute(s.Range), s.Ens.ConvertAll(SubstMayBeFreeExpr), body);
           rr.Kind = s.Kind;
+          rr.CanConvert = s.CanConvert;
+          rr.Bounds = s.Bounds; // not substituted (should it be?)
+          if (s.ForallExpressions != null) {
+            var list = new List<Expression>();
+            foreach (Expression e in s.ForallExpressions) {
+              list.Add(Substitute(e));
+            }
+            rr.ForallExpressions = list;
```

Comment:
A more concise way to write this is

    rr.ForallExpression.ConvertAll(Substitute);

Reply:
OK